mne/io/curry/curry.py

```diff
@@ -191,39 +225,76 @@ def _read_curry_info(curry_paths):
+                chanidx = int(curry_params.chanidx_in_file["CHAN_IN_FILE"
+                              + CHANTYPES[key]][ind])
+            if chanidx > 0:   # if chanidx was explicitly declared to be ' 0',
+                # it means the channel is not actually saved in the data file
+                # (e.g. the "Ref" channel), so don't add it to our list.
+                # Git issue #8391
+                ch = {"ch_name": chan,
+                      "unit": curry_params.unit_dict[key],
+                      "kind": FIFFV_CHANTYPES[key],
+                      "coil_type": FIFFV_COILTYPES[key],
+                      "ch_idx": chanidx
+                      }
+                if key == "eeg":
+                    loc = np.array(sensors["SENSORS" + CHANTYPES[key]][ind],
+                                   float)
+                    # XXX just the sensor, where is ref (next 3)?
+                    assert loc.shape == (3,)
+                    loc /= 1000.  # to meters
+                    loc = np.concatenate([loc, np.zeros(9)])
+                    ch['loc'] = loc
+                    # XXX need to check/ensure this
+                    ch['coord_frame'] = FIFF.FIFFV_COORD_HEAD
+                elif key == 'meg':
+                    pos = np.array(sensors["SENSORS" + CHANTYPES[key]][ind],
+                                   float)
+                    pos /= 1000.  # to meters
+                    pos = pos[:3]  # just the inner coil
+                    pos = apply_trans(curry_dev_dev_t, pos)
+                    nn = np.array(normals["NORMALS" + CHANTYPES[key]][ind],
+                                  float)
+                    assert np.isclose(np.linalg.norm(nn), 1., atol=1e-4)
+                    nn /= np.linalg.norm(nn)
+                    nn = apply_trans(curry_dev_dev_t, nn, move=False)
+                    trans = np.eye(4)
+                    trans[:3, 3] = pos
+                    trans[:3, :3] = _normal_orth(nn).T
+                    ch['loc'] = _coil_trans_to_loc(trans)
+                    ch['coord_frame'] = FIFF.FIFFV_COORD_DEVICE
+                all_chans.append(ch)
+
+    ch_count = len(all_chans)
+    assert (ch_count == curry_params.n_chans)  # ensure that we have assembled
+    # the same number of channels as declared in the info (.DAP) file in the
+    # DATA_PARAMETERS section. Git issue #8391
+
+    # sort the channels to assure they are in the order that matches how
+    # recorded in the datafile.  In general they most likely are already in
+    # the correct order, but if the channel index in the data file was
+    # explicitly declared we might as well use it.  This is a simple bubble
+    # sort.
+    for i in range(0, ch_count):
+        for j in range(0, ch_count - i - 1):
+            if (all_chans[j]["ch_idx"] > all_chans[j + 1]["ch_idx"]):
+                temp = all_chans[j]
+                all_chans[j] = all_chans[j + 1]
+                all_chans[j + 1] = temp
```

Comment:
```suggestion
    all_chans = sorted(all_chans, key=lambda ch: ch['ch_idx'])
```

Reply:
Excellent!  I would have done a lambda function like that in .Net, but had no idea how to do it in Python!  Thank you!